### PR TITLE
fix RIDER-26102, RIDER-26109, RIDER-26103 

### PR DESCRIPTION
**Pull request by van800**

### CODE REVIEW

File: resharper/resharper-unity/src/ProjectExtensions.cs [hunk 1]
```diff
@@ -40,7 +40,7 @@ public static bool HasUnityReference([NotNull] this ISolution solution)
         public static bool IsUnityProject([CanBeNull] this IProject project)
         {
             // Only VSTU adds the Unity project flavour. Unity + Rider don't, so we have to look at references
-            return project != null && (project.HasFlavour<UnityProjectFlavor>() || ReferencesUnity(project));
+            return project != null && (project.HasFlavour<UnityProjectFlavor>() || project.IsValid() && ReferencesUnity(project));
```
Comment (citizenmatt) [inline resharper/resharper-unity/src/ProjectExtensions.cs]: Why would `project` be invalid here?

Comment (van800) [reply resharper/resharper-unity/src/ProjectExtensions.cs]: missing/incompatible msbuild?

Comment (citizenmatt) [reply resharper/resharper-unity/src/ProjectExtensions.cs]: Oh, would we still have a project in that case?

Comment (van800) [reply resharper/resharper-unity/src/ProjectExtensions.cs]: I don't know, I can't reproduce it for real

**Pull request merged**